partiql-types/src/test/kotlin/org/partiql/datetime/TimestampTest.kt

```diff
@@ -0,0 +1,918 @@
+            ),
+            SuccessCaseWithNoTimeZone(
+                "Rounding - up",
+                1970, 1, 1, 0, 0, BigDecimal.valueOf(999999, 5), 0,
+                DateTimeValue.timestamp(
+                    DateTimeValue.date(1970, 1, 1),
+                    DateTimeValue.time(0, 0, BigDecimal.valueOf(10, 0),)
+                ),
+            ),
+            SuccessCaseWithNoTimeZone(
+                "Rounding - carry to minute",
+                1970, 1, 1, 0, 0, BigDecimal.valueOf(599, 1), 0,
+                DateTimeValue.timestamp(
+                    DateTimeValue.date(1970, 1, 1),
+                    DateTimeValue.time(0, 1, BigDecimal.valueOf(0, 0),)
+                ),
+            ),
+            SuccessCaseWithNoTimeZone(
+                "Rounding - carry to hour",
+                1970, 1, 1, 0, 59, BigDecimal.valueOf(599, 1), 0,
+                DateTimeValue.timestamp(
+                    DateTimeValue.date(1970, 1, 1),
+                    DateTimeValue.time(1, 0, BigDecimal.valueOf(0, 0),)
+                ),
+            ),
+            SuccessCaseWithNoTimeZone(
+                "Rounding - carry to day",
+                1970, 1, 1, 23, 59, BigDecimal.valueOf(599, 1), 0,
+                DateTimeValue.timestamp(
+                    DateTimeValue.date(1970, 1, 2),
+                    DateTimeValue.time(0, 0, BigDecimal.valueOf(0, 0),)
+                ),
+            ),
+            SuccessCaseWithNoTimeZone(
+                "Rounding - carry to month",
+                1970, 1, 31, 23, 59, BigDecimal.valueOf(599, 1), 0,
+                DateTimeValue.timestamp(
+                    DateTimeValue.date(1970, 2, 1),
+                    DateTimeValue.time(0, 0, BigDecimal.valueOf(0, 0),)
+                ),
+            ),
+            SuccessCaseWithNoTimeZone(
+                "Rounding - carry to year",
+                1970, 12, 31, 23, 59, BigDecimal.valueOf(599, 1), 0,
+                DateTimeValue.timestamp(
+                    DateTimeValue.date(1971, 1, 1),
+                    DateTimeValue.time(0, 0, BigDecimal.valueOf(0, 0),)
+                ),
+            ),
+            SuccessCaseWithNoTimeZone(
+                "Large precision",
+                1970, 1, 1, 0, 0, BigDecimal.valueOf(0, 20), 20,
+                DateTimeValue.timestamp(
+                    DateTimeValue.date(1970, 1, 1),
+                    DateTimeValue.time(0, 0, BigDecimal.valueOf(0, 20),)
+                ),
+            ),
+            // negative epoch
+            // May be Ion has a bug?
+            // Expected epoch in Millis : -23225270400000,
+            // Actual epoch in Millis   : -23225875200000
+            // SuccessCaseWithNoTimeZone(
+            //    1234, 1, 1, 0,0, BigDecimal.valueOf(0,3), 0,
+            //    Timestamp.of(
+            //        DateTimeValue.date(1234, 1, 1),
+            //        DateTimeValue.time(0,0, BigDecimal.valueOf(0,0), )
+            // ),
+            // ),
```

Comment:
While the matching `SuccessCaseWithUnknownTimeZone` fails -- would this one also fail? This specific test case shouldn't be comparing Ion values if I'm reading this correctly.